google-cloud-firestore/src/main/java/com/google/cloud/firestore/RateLimiter.java

```diff
@@ -38,31 +38,48 @@
   private final double multiplier;
   private final int multiplierMillis;
   private final long startTimeMillis;
+  private final int maximumCapacity;
 
   private int availableTokens;
   private long lastRefillTimeMillis;
 
-  RateLimiter(int initialCapacity, double multiplier, int multiplierMillis) {
-    this(initialCapacity, multiplier, multiplierMillis, new Date().getTime());
+  RateLimiter(int initialCapacity, double multiplier, int multiplierMillis, int maximumCapacity) {
+    this(initialCapacity, multiplier, multiplierMillis, maximumCapacity, new Date().getTime());
   }
 
   /**
    * @param initialCapacity Initial maximum number of operations per second.
    * @param multiplier Rate by which to increase the capacity.
    * @param multiplierMillis How often the capacity should increase in milliseconds.
+   * @param maximumCapacity Maximum number of allowed operations per second. The number of tokens
```

Comment:
Isn't this different from the capacity? From what I can tell, the capacity grows unbounded and the rate is reduced artificially. The implementation is fine, but the name of the setting is a bit misleading.

Reply:
When the capacity is calculated, it's capped at the `maximumCapacity`, which I thought is short for "maximum allowed capacity". 

When a new request is made, the number of tokens to allocate is done in `calculateCapacity`, which performs the `Math.min()` operation. This means that the capacity is bounded by `maximumCapacity`.

Reply:
I think in your implementation the total capacity can grow beyond `maximumCapacity`, but the rate at which tokens get deducted is limited. This implementation is correct, but the value that is passed here is not the maximum capacity, but rather the limit of tokens used at a time. e.g. after 5 seconds without requests the total capacity of the throttler could be 500, even if `maximumCapacity` is 100. 

As stated, this is just a naming nit.

Reply:
Thanks for elaborating! Renamed to `maximumRate`.